src/python/pants/backend/python/lint/ruff/rules.py

```diff
@@ -64,8 +71,21 @@ async def run_ruff(
     )
 
     conf_args = [f"--config={ruff.config}"] if ruff.config else []
+
+    extra_initial_args: Tuple[str, ...] = ()
+    if request.mode == RuffMode.FORMAT:
+        extra_initial_args = ("format",)
+    elif request.mode == RuffMode.FIX:
+        # TODO: In ruff >= 0.1 this should be ("check", "--fix"), but we retain
+        # the old behavior for backwards compatibility. Change this when support
+        # for ruff < 0.1 is dropped.
+        # See https://docs.astral.sh/ruff/linter/ for more details.
+        extra_initial_args = ("--fix",)
```

Comment:
Idk if anyone is using Ruff with a custom tool lockfile - if they are, then this would break their use case. But if we don't particularly care about that then I'm happy to just fix this now.

Reply:
Mulled over this a bit and given that Ruff was previously a 0.0.x version tool, I don't think there is a reasonable expectation of stability. Updated to address the comment.

Reply:
I appreciate you thought of this, and also we normally want to be conservative with breaking changes going through deprecation cycles etc. But as this is still an experimental backend we're not obliged to follow our normal deprecation policies in order to make iterative improvements easier.